### DISCUSSION TITLE
v7.3.0-preview.8 Release of PowerShell

### DISCUSSION THREAD

**Discussion by daxian-dbw** [PowerShell/PowerShell#18135] [Show and tell]
## [7.3.0-preview.8] - 2022-09-20

**NOTE:** Known issue about `Console.ReadKey()` on Unix platforms that the `KeyChar` is not preserved for <kbd>Ctrl+Letter</kbd> inputs. This causes all key bindings in the form of <kbd>Ctrl+Letter</kbd> to not work in PowerShell `v7.3.0-preview.8` on Unix platforms. For details, see the .NET issue https://github.com/dotnet/runtime/issues/75795. The fix https://github.com/dotnet/runtime/pull/75853 will be included in .NET `7.0.0-rc.2` and we plan to include this in the next PowerShell `7.3` release.

### General Cmdlet Updates and Fixes

- Filter out compiler generated types for `Add-Type -PassThru` (#18095)
- Fix error formatting to use color defined in `$PSStyle.Formatting` (#17987)
- Handle `PSObject` argument specially in method invocation logging (#18060)
- Revert the experimental feature `PSStrictModeAssignment` (#18040)
- Make experimental feature `PSAMSIMethodInvocationLogging` stable (#18041)
- Make experimental feature `PSAnsiRenderingFileInfo` stable (#18042)
- Make experimental feature `PSCleanBlock` stable (#18043)
- Make experimental feature `PSNativeCommandArgumentPassing` stable (#18044)
- Make experimental feature `PSExec` stable (#18045)
- Make experimental feature `PSRemotingSSHTransportErrorHandling` stable (#18046)
- Add the `ConfigurationFile` option to the PowerShell help content (#18093)

### Build and Packaging Improvements

<summary>
<p>Bump .NET SDK to version <code>7.0.100-rc.1</code></p>
</summary>

<details>
<ul>
<li>Update ThirdPartyNotices.txt for 7.3.0-preview.8 (Internal 22553)</li>
<li>Update cgmanifest.json for 7.3.0-preview.8 (Internal 22551)</li>
<li>Re-enable building with Ready-to-Run (#18107)</li>
<li>Make sure <code>Security.types.ps1xml</code> gets signed in release build (#17930)</li>
<li>Update <code>DotnetRuntimeMetadata.json</code> for .NET 7 RC1 build (#18106)</li>
<li>Add XML reference documents to NuPkg files for SDK (#18017)</li>
<li>Make Register MU timeout (#17995)</li>
<li>Bump Microsoft.NET.Test.Sdk from 17.2.0 to 17.3.0 (#17924)</li>
<li>Update list of PS team members in release tools (#17928)</li>
<li>Update to use version 2.21.0 of Application Insights (#17927)</li>
<li>Complete ongoing <code>Write-Progress</code> in test (#17922)</li>
</ul>
</details>

[7.3.0-preview.8]: https://github.com/PowerShell/PowerShell/compare/v7.3.0-preview.7...v7.3.0-preview.8

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - B8FE62FE51C391395CD43E3411B73885E7F58C671506243161D742F39597DF78
- powershell-7.3.0-preview.8-linux-alpine-x64.tar.gz
  - 9F25E94EA30DAA89BD17D3A35D753138D02F829D223ABAC4D9E1E116A7D4648F
- powershell-7.3.0-preview.8-linux-arm32.tar.gz
  - 6FC64E841EF3DC701E99FA58FFD4502FD6EE8F8C3FCED1A2FD5F917A72AB6483
- powershell-7.3.0-preview.8-linux-arm64.tar.gz
  - 503A624D4B880F8D86FB52105D1680EA833278546D07052BD6EB82E5413B3886
- powershell-7.3.0-preview.8-linux-x64.tar.gz
  - B6F3EDF099475012284A27455D49609A8A0CAA5A35FBD5AF1D2FBEE65A3F539A
- powershell-7.3.0-preview.8-linux-x64-fxdependent.tar.gz
  - F3B87F93EFF7861AFD87EC53801A62D27ACABE5F081AA106D9FC65B5A32A9E3B
- powershell-7.3.0-preview.8-osx-arm64.pkg
  - 6A7E3B19A6282931230A56F56A18F99C3F582ACB30985AED47D4B70192584605
- powershell-7.3.0-preview.8-osx-arm64.tar.gz
  - 1D9A16A998A2D36955285BF9DE65E45F69EFB4067E54C120B6F96AA224459CE1
- powershell-7.3.0-preview.8-osx-x64.pkg
  - 84AC26DA95486D0BB417BE13C7C5E901A1A1AC7F4F756EF8454376E2C40860CC
- powershell-7.3.0-preview.8-osx-x64.tar.gz
  - 7F1432AD05D48487C8B4A5A2652CF4360DCEBC05CC30FF129E939EBB18B29211
- PowerShell-7.3.0-preview.8-win-arm32.zip
  - 41854CD7D6283E3EDC3CA5836E3F9D7460E287F8CCD6AFB540288F8CD9B70593
- PowerShell-7.3.0-preview.8-win-arm64.zip
  - 4D85D9F3385050ABAA7D60ABEF6B1BF1EB347159269F1A140B7F652630D99255
- PowerShell-7.3.0-preview.8-win-fxdependent.zip
  - 195E46DFF99DBE778C602D600933D8895AFFEBFE314F5419699215EBA48057F3
- PowerShell-7.3.0-preview.8-win-fxdependentWinDesktop.zip
  - CEFDB7DE0447CBD21FBD576FCBB9F2A14AC41340817F4E9F94E0B5E9C5CE3D1B
- PowerShell-7.3.0-preview.8-win-x64.msi
  - A97001262EECD33EB210F670AC8C43DB2E46CEED525481F1EA724F0578CCCD68
- PowerShell-7.3.0-preview.8-win-x64.zip
  - 083CF35701E5F25FBB09A8CB24E956E327EB7B1179ACD21FB0F5479782F9E9C9
- PowerShell-7.3.0-preview.8-win-x86.msi
  - 1370E020AE3EFF3BF0905E415EAB21679B7767464AC3DF0098F60736F5CD5813
- PowerShell-7.3.0-preview.8-win-x86.zip
  - 3F100E348414FBB345E0349ED44FFE5C6090503793E3609A7D296E0D76F9F507
- powershell-preview_7.3.0-preview.8-1.deb_amd64.deb
  - 7A3AEE968950BE1B9DC7E8900CE2AC8537CC349DBB5F8C9EC4147AE5D6C704A1
- powershell-preview-7.3.0_preview.8-1.cm.x86_64.rpm
  - 30F4329AD0B8C9EAAC23F4FDB796937AC912D8238313DFD4C47132C845FA23CC
- powershell-preview-7.3.0_preview.8-1.rh.x86_64.rpm
  - FBD9DF4B6BD4273C4B102375FF3C2B9D760AD0FC40AD723CABE7277BDA0055CA

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.3.0-preview.8'>v7.3.0-preview.8 Release of PowerShell</a>.</em>

**Comment by berot3** [comment]
how long until this lands on homebrew? 😅

**Comment by daxian-dbw** [comment]
https://github.com/PowerShell/Homebrew-Tap/pull/960 has been merged, so it should be available on Homebrew-Tap already. See https://github.com/PowerShell/Homebrew-Tap#homebrew-tap-for-powershell-products for the installation instructions.

**Comment by iSazonov** [comment]
There is again an issue with git tag. I did not get latest one automatically. After manual sync it doesn't work still - Start-PSBuild continue build Preview.3 (coming from git describe).
